src/test/java/com/commercetools/sync/cartdiscounts/utils/CartDiscountUpdateActionUtilsTest.java

```diff
@@ -0,0 +1,563 @@
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        final CartDiscountTarget cartDiscountTarget = LineItemsTarget.of("quantity > 0");
+        when(newCartDiscountDraft.getTarget()).thenReturn(cartDiscountTarget);
+
+        final Optional<UpdateAction<CartDiscount>> changeTargetUpdateAction =
+            buildChangeTargetUpdateAction(CART_DISCOUNT_WITH_RELATIVE_VALUE, newCartDiscountDraft);
+
+        assertThat(changeTargetUpdateAction).contains(ChangeTarget.of(cartDiscountTarget));
+    }
+
+    @Test
+    public void buildChangeTargetUpdateAction_WithSameLineItemTargetValues_ShouldNotBuildUpdateAction() {
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        final CartDiscountTarget cartDiscountTarget = LineItemsTarget.of("1 = 1");
+        when(newCartDiscountDraft.getTarget()).thenReturn(cartDiscountTarget);
+
+        final Optional<UpdateAction<CartDiscount>> changeTargetUpdateAction =
+            buildChangeTargetUpdateAction(CART_DISCOUNT_WITH_RELATIVE_VALUE, newCartDiscountDraft);
+
+        assertThat(changeTargetUpdateAction).isNotPresent();
+    }
+
+    @Test
+    public void buildChangeTargetUpdateAction_WithNullValues_ShouldBuildUpdateAction() {
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        when(newCartDiscountDraft.getTarget()).thenReturn(null);
+
+        final Optional<UpdateAction<CartDiscount>> changeTargetUpdateAction =
+            buildChangeTargetUpdateAction(CART_DISCOUNT_WITH_RELATIVE_VALUE, newCartDiscountDraft);
+
+        assertThat(changeTargetUpdateAction).contains(ChangeTarget.of(null));
+    }
+
+    @Test
+    public void buildChangeTargetUpdateAction_WithDifferentCustomLineItemValues_ShouldBuildUpdateAction() {
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        final CartDiscountTarget cartDiscountTarget = CustomLineItemsTarget.of("1 = 1");
+        when(newCartDiscountDraft.getTarget()).thenReturn(cartDiscountTarget);
+
+        final Optional<UpdateAction<CartDiscount>> changeTargetUpdateAction =
+            buildChangeTargetUpdateAction(CART_DISCOUNT_WITH_CUSTOM_LINE_ITEM_TARGET, newCartDiscountDraft);
+
+        assertThat(changeTargetUpdateAction).contains(ChangeTarget.of(cartDiscountTarget));
+    }
+
+    @Test
+    public void buildChangeTargetUpdateAction_WithLineItemAndShippingTargetValues_ShouldBuildUpdateAction() {
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        final CartDiscountTarget cartDiscountTarget = ShippingCostTarget.of();
+        when(newCartDiscountDraft.getTarget()).thenReturn(cartDiscountTarget);
+
+        final Optional<UpdateAction<CartDiscount>> changeTargetUpdateAction =
+            buildChangeTargetUpdateAction(CART_DISCOUNT_WITH_RELATIVE_VALUE, newCartDiscountDraft);
```

Comment:
also here: https://github.com/commercetools/commercetools-sync-java/pull/388/files#diff-30e0149be0301bb3ff8a918ecc1b08d5R153

Reply:
There are many cases which are not covered by the unit tests:

`MultiBuyCustomLineItemsTarget`
`MultiBuyLineItemsTarget`

with all the variation of their sub fields! :(

Reply:
f8651bc